### PR TITLE
MDS-4174 - Nod upload/view/download

### PR DESCRIPTION
**Pull request by matbusby-fw**
## Objective 

- Added `fileUpload` component to `Create Notice of Departure`
- Added 2 new redux action creators `fetchDetailedNoticeOfDeparture` and `addDocumentToNoticeOfDeparture`
- Added reducer for storing/retrieving detailed NOD
- Updated view NOD modal styling to match designs (previously built off of wireframe)

[MDS-](https://bcmines.atlassian.net/browse/MDS-####)

## Additional Information / Context 

image
image

### CODE REVIEW

File: services/minespace-web/src/components/Forms/noticeOfDeparture/AddNoticeOfDepartureForm.js [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 import { Button, Col, Popconfirm, Row, Typography } from "antd";
 import { Form } from "@ant-design/compatible";
 import { maxLength, required, requiredList, validateSelectOptions } from "@common/utils/Validate";
 import { resetForm } from "@common/utils/helpers";
+import { DOCUMENT, EXCEL } from "@/constants/fileTypes";
 import { renderConfig } from "@/components/common/config";
 import * as FORM from "@/constants/forms";
 import CustomPropTypes from "@/customPropTypes";
+import NoticeOfDepartureFileUpload from "@/components/Forms/noticeOfDeparture/NoticeOfDepartureFileUpload";
+import { NOTICE_OF_DEPARTURE_DOCUMENT_TYPE } from "../../../../common/constants/strings";
```
Comment (hitankar-freshworks) [inline services/minespace-web/src/components/Forms/noticeOfDeparture/AddNoticeOfDepartureForm.js]: shouldn't this be `@common/constants/strings` right?

File: services/minespace-web/src/components/dashboard/mine/noticeOfDeparture/NoticeOfDepartureDetails.js [hunk 2]
```diff
@@ -1,24 +1,113 @@
...
+import { Col, Divider, Row } from "antd";
 import CustomPropTypes from "@/customPropTypes";
+import { NOTICE_OF_DEPARTURE_DOCUMENT_TYPE } from "../../../../../common/constants/strings";
+import {
+  EMPTY_FIELD,
+  NOTICE_OF_DEPARTURE_STATUS,
+  NOTICE_OF_DEPARTURE_TYPE,
+} from "@/constants/strings";
+import LinkButton from "@/components/common/LinkButton";
+import { downloadFileFromDocumentManager } from "../../../../../common/utils/actionlessNetworkCalls";
```
Comment (hitankar-freshworks) [inline services/minespace-web/src/components/dashboard/mine/noticeOfDeparture/NoticeOfDepartureDetails.js]: should it be `@common...`?

File: services/minespace-web/src/styles/components/NoticeOfDeparture.scss [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+.nod-section-padding {
+  padding: 11px;
+}
+
+.nod-cancel-button {
+  transition: all 0.1s ease-in-out;
+  background: #FFFFFF;
+  //border: 2px solid $violet;
```
Comment (hitankar-freshworks) [inline services/minespace-web/src/styles/components/NoticeOfDeparture.scss]: probably remove commented code

File: services/core-api/app/api/mines/notice_of_departure/resources/notice_of_departure.py [hunk 4]
```diff
@@ -1,23 +1,19 @@
...
-from app.api.mines.mine.models.mine import Mine
-from app.api.mines.response_models import PERMIT_NOD_MODEL
+from app.api.mines.response_models import NOD_MODEL
 
 
 class NoticeOfDepartureResource(Resource, UserMixin):
 
-    @api.doc(params={
-        'mine_guid': 'Mine guid.',
-        'nod_guid': 'Mine guid.'
```
Comment (matbusby-fw) [inline services/core-api/app/api/mines/notice_of_departure/resources/notice_of_departure.py:17]: I think this is likely not a big deal, but shouldn't `Mine guid` on line 17 be `Nod Guid`? @hitankar-freshworks

**Review by hitankar-freshworks — DISMISSED** [review]
Looks good except couple of comments

File: services/minespace-web/src/components/Forms/noticeOfDeparture/AddNoticeOfDepartureForm.js (hunk 1 above)
Comment (matbusby-fw) [reply services/minespace-web/src/components/Forms/noticeOfDeparture/AddNoticeOfDepartureForm.js]: Good catch.  The IDE sometimes grabs them from their path automatically and adds those imports.  Didn't see it happen!

File: services/core-api/app/api/mines/notice_of_departure/resources/notice_of_departure.py (hunk 4 above)
Comment (hitankar-freshworks) [reply services/core-api/app/api/mines/notice_of_departure/resources/notice_of_departure.py:17]: Thats right! if you look down. You will see that is already updated

File: services/minespace-web/src/components/dashboard/mine/noticeOfDeparture/NoticeOfDepartureDetails.js [hunk 5]
```diff
@@ -1,15 +1,15 @@
...
-import { NOTICE_OF_DEPARTURE_DOCUMENT_TYPE } from "../../../../../common/constants/strings";
 import {
   EMPTY_FIELD,
   NOTICE_OF_DEPARTURE_STATUS,
   NOTICE_OF_DEPARTURE_TYPE,
 } from "@/constants/strings";
 import LinkButton from "@/components/common/LinkButton";
-import { downloadFileFromDocumentManager } from "../../../../../common/utils/actionlessNetworkCalls";
 import { formatDate } from "@/utils/helpers";
+import { NOTICE_OF_DEPARTURE_DOCUMENT_TYPE } from "../../../../../common/constants/strings";
```
Comment (hitankar-freshworks) [inline services/minespace-web/src/components/dashboard/mine/noticeOfDeparture/NoticeOfDepartureDetails.js]: @matbusby-fw This one

Comment (matbusby-fw) [reply services/minespace-web/src/components/dashboard/mine/noticeOfDeparture/NoticeOfDepartureDetails.js]: :man_facepalming:  Thanks!

**Review by hitankar-freshworks — APPROVED** [review]

**Pull request merged**